### PR TITLE
Settlement totals compare function

### PR DESCRIPTION
**Pull request by SiFalorSum**

### CODE REVIEW

File: Settlement/Total.ts [hunk 1]
```diff
@@ -14,6 +15,16 @@ export namespace Total {
...
+	export function compare(expected: Total, outcome: Total): boolean {
+		return (
+			([...new Set([...Object.keys(expected.amount), ...Object.keys(outcome.amount)])] as isoly.Currency[]).every(
+				currency => expected.amount[currency] == outcome.amount[currency]
+			) &&
+			([...new Set([...Object.keys(expected.fee.other), ...Object.keys(outcome.fee.other)])] as isoly.Currency[]).every(
+				currency => expected.fee.other[currency] == outcome.fee.other[currency]
+			)
+		)
+	}
```
Comment (marcushanses) [inline Settlement/Total.ts:27]: It's simpler to use Object.entries(expected.amount).every(([currency, amount] => outcome.amount[currency] == amount))

Comment (marcushanses) [reply Settlement/Total.ts:27]: Also a function which is typically good to have a test for. (You don't have to but would be nice.

**Review by marcushanses — CHANGES_REQUESTED** [review]

File: Settlement/Total.ts (hunk 1 above)
Comment (SiFalorSum) [reply Settlement/Total.ts:27]: I'm trying to cover for when "expected" might have fewer currencies than "outcome", but maybe that requires some different error handling?

Comment (marcushanses) [reply Settlement/Total.ts:27]: Fair enough.

**Review by marcushanses — APPROVED** [review]

**Pull request merged**